lib/iex/lib/iex/introspection.ex

```diff
@@ -39,112 +30,183 @@ defmodule IEx.Introspection do
   @doc """
   Prints the documentation for the given function
   with any arity in the list of modules.
+
+  Used to find functions that either iex commands or kernel functions.
   """
   def h(modules, function) when is_list(modules) and is_atom(function) do
-    result =
-      Enum.find_value modules, false, fn module ->
-        h_mod_fun(module, function) == :ok
-      end
-
-    unless result, do: nodocs(function)
+    helpers = IEx.Config.doc_helpers(:helpers)
+    result = Enum.find_value(modules,
+               fn(module) ->
+                  Enum.find_value(helpers,
+                        fn(mod) -> found_help(mod, module, function) end)
+               end)
```

Comment:
thank you @whatyouhide that's a really neat solution. I didn't know about that one, but look clean and readable.